pwiz_tools/Skyline/Util/ModeUIAwareForm.cs

```diff
@@ -257,15 +279,16 @@ internal void AdjustMenusForModeUI(ToolStripItemCollection items)
                 }
             }
 
-            public bool MenuItemHasOriginalText(string name)
+            public bool MenuItemHasOriginalText(ToolStripMenuItem toolStripMenuItem)
             {
                 foreach (var item in _modeUIExtender.GetOriginalToolStripText().Keys)
                 {
                     if (item is ToolStripMenuItem menuItem)
                     {
-                        if (Equals(name, menuItem.Text))
+                        if (Equals(toolStripMenuItem.Text, menuItem.Text) && Equals(toolStripMenuItem.ToolTipText, menuItem.ToolTipText))
```

Comment:
You could take advantage of the fact that the struct "MenuTextAndToolTip" which gives you .Equals for free.
```
  public bool MenuItemHasOriginalText(ToolStripMenuItem toolStripMenuItem)
  {
      var textAndTooltip = new ModeUIExtender.MenuTextAndToolTip(toolStripMenuItem);
      foreach (var keyValuePair in _modeUIExtender.GetOriginalToolStripText())
      {
          if (keyValuePair.Key is ToolStripMenuItem menuItem)
          {
              if (Equals(new ModeUIExtender.MenuTextAndToolTip(menuItem), textAndTooltip))
              {
                  return Equals(keyValuePair.Value, textAndTooltip);
              }
          }
      }
      return false;
  }

```

Reply:
Interesting thought. The only disadvantage to that approach is that it uses reflection to implement the memberwise comparisons, so it's slower than the hand rolled version. Probably doesn't really matter in this case, though. At any rate this works and is just as easy to understand so I'm going to leave it alone.